src/editor/Collaboration.js

```diff
@@ -160,18 +166,136 @@ define(function (require, exports, module) {
     };
 
     function _handleFileChangeEvent(path, change) {
-        var file = FileSystem.getFileForPath(path);
-        console.log("Should write to file which is not open in editor." + file + "changed " + change);
+        if(!_buffer[path]) {
+            _buffer[path] = [];
+        }
+        _buffer[path].push(change);
     };
 
+    function _clearBuffer() {
+        for(var path in _buffer) {
+            if(_buffer[path].length > 0) {
+                clearFile(path);
+            }
+        }
+    }
+
+    /**
+     * Applies all the changes kept in the buffer array that have occured on connected clients but have not
+     * yet been written to the filesystem for this file.
+     *
+     * @param {!string} path
+     * @return {$.Promise} A promise object that will be resolved when the buffer array for the file
+     * is cleared or rejected with a FileSystemError if the file is not yet open and can't be read from disk.
+     */
+    function clearFile(path) {
+        var result = new $.Deferred();
+        var file = FileSystem.getFileForPath(path);
+        if(!_webrtc || !_buffer || !_buffer[path] || _buffer[path].length === 0) {
+            FileUtils.readAsText(file)
```

Comment:
You're not doing anything special here so you don't need any of the promise logic. Instead, this should actually just be:
```js
return FilerUtils.readAsUTF8(file);
```